src/Microsoft.Fast.Components.FluentUI/Components/InputFile/FluentInputFile.razor.cs

```diff
@@ -0,0 +1,338 @@
+
+        await Module.InvokeVoidAsync("raiseFluentInputFile", Id);
+    }
+
+    /// <summary />
+    protected async Task OnUploadFilesHandlerAsync(InputFileChangeEventArgs e)
+    {
+        if (e.FileCount > MaximumFileCount)
+        {
+            throw new ApplicationException($"The maximum number of files accepted is {MaximumFileCount}, but {e.FileCount} were supplied.");
+        }
+
+        DropOver = false;
+
+        // Use the native Blazor event
+        if (OnInputFileChange.HasDelegate)
+        {
+            await OnInputFileChange.InvokeAsync(e);
+            return;
+        }
+
+        // Start
+        ProgressPercent = 0;
+        ProgressTitle = FluentInputFileResource.LoadingBefore;
+
+        List<FluentInputFileEventArgs>? uploadedFiles = new();
+        IReadOnlyList<IBrowserFile>? allFiles = e.GetMultipleFiles(MaximumFileCount);
+        (string Name, long Size, string ContentType)[]? allFilesSummary = allFiles.Select(i => (i.Name, i.Size, i.ContentType)).ToArray();
+        long totalFileSizes = allFiles.Sum(i => i.Size);
+        long totalRead = 0L;
+        int fileNumber = 0;
+
+
+
+        foreach (IBrowserFile file in allFiles)
+        {
+            // Keep a trace of this file
+            FluentInputFileEventArgs? fileDetails = new FluentInputFileEventArgs()
+            {
+                AllFiles = allFilesSummary,
+                Index = fileNumber,
+                Name = file.Name,
+                ContentType = file.ContentType,
+                Size = file.Size,
+                IsCancelled = false,
+            };
+            uploadedFiles.Add(fileDetails);
+
+            // Max size => ERROR
+            if (file.Size > MaximumFileSize)
+            {
+                fileDetails.ErrorMessage = "The maximum size allowed is reached";
+                continue;
+            }
+
+            // Progress
+            ProgressTitle = string.Format(FluentInputFileResource.LoadingInProgress, fileNumber + 1, allFiles.Count, file.Name);
+            fileDetails.ProgressTitle = ProgressTitle;
+
+
+            switch (Mode)
+            {
+                case InputFileMode.Buffer:
+
+                    // Read all buffers and raise the event OnProgressChange
+                    await ReadFileToBufferAndRaiseProgressEventAsync(file, fileDetails, (buffer, bytesRead) =>
+                    {
+                        totalRead += bytesRead;
+
+                        fileDetails.Buffer = new FluentInputFileBuffer(buffer, bytesRead);
+
+                        ProgressPercent = Convert.ToInt32(decimal.Divide(totalRead, totalFileSizes) * 100);
+                        return Task.CompletedTask;
+                    });
+
+                    break;
+
+                case InputFileMode.SaveToTemporaryFolder:
+
+                    // Save to temporary file
+                    string? tempFileName = Path.GetTempFileName();
```

Comment:
Did you check this method in WASM mode? Is it possible to create a Temp File?

Reply:
Yes, that works as well,